contractcourt/briefcase.go

```diff
@@ -469,50 +469,35 @@ func (b *boltArbitratorLog) FetchUnresolvedContracts() ([]ContractResolver, erro
 
 			switch resType {
 			case resolverTimeout:
-				timeoutRes := &htlcTimeoutResolver{}
-				if err := timeoutRes.Decode(resReader); err != nil {
+				res, err = decodeTimeoutResolver(resReader)
+				if err != nil {
 					return err
 				}
-				timeoutRes.AttachResolverKit(resKit)
-
-				res = timeoutRes
 
 			case resolverSuccess:
-				successRes := &htlcSuccessResolver{}
-				if err := successRes.Decode(resReader); err != nil {
+				res, err = decodeSuccessResolver(resReader)
+				if err != nil {
 					return err
 				}
 
-				res = successRes
-
 			case resolverOutgoingContest:
-				outContestRes := &htlcOutgoingContestResolver{
-					htlcTimeoutResolver: htlcTimeoutResolver{},
-				}
-				if err := outContestRes.Decode(resReader); err != nil {
+				res, err = decodeOutgoingContestResolver(resReader)
+				if err != nil {
 					return err
 				}
 
-				res = outContestRes
-
 			case resolverIncomingContest:
-				inContestRes := &htlcIncomingContestResolver{
-					htlcSuccessResolver: htlcSuccessResolver{},
-				}
-				if err := inContestRes.Decode(resReader); err != nil {
+				res, err = decodeIncomingContestResolver(resReader)
```

Comment:
What do these methods really achieve, that `Decode` doesn't?

Reply:
The end result is the same, but it encapsulated the instantiation of the resolver inside the resolver. If we want to follow up with additional initialization of resolvers (for example set up the initial report), it is clear where to do it. Otherwise that init code will need to be called separately from chan arb which carries more risk of making a mistake.